File: EosSocial/wm_inspect.py
from gi.repository import GObject
from gi.repository import Wnck


class WM_Inspect_MixIn(object):


    def __init__(self):
        super(WM_Inspect_MixIn, self).__init__()

        self._screen = Wnck.Screen.get_default()
        self._screen.connect('active-window-changed', self._on_active_window_changed)

    def _on_active_window_changed(self, previous, data):
        active_window = self._screen.get_active_window()
        if not active_window:
            return
        if hasattr(self, '_active_win_callback'):
            self._active_win_callback(active_window.get_xid())

File: EosSocial/wmInspect.js
// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-

const Lang = imports.lang;
const Signals = imports.signals;
const Wnck = imports.gi.Wnck;

const WMInspect = new Lang.Class({
    Name: 'WMInspect',

    _init: function() {
        this._screen = Wnck.Screen.get_default();
        this._screen.connect('active-window-changed', Lang.bind(this,
            this._onActiveWindowChanged));
    },

    _onActiveWindowChanged: function() {
        let activeWindow = this._screen.get_active_window();

        if (!activeWindow)
            return;

        this.emit('active-window-changed', activeWindow);
    }
});
Signals.addSignalMethods(WMInspect.prototype);
